### PR TITLE
Upload standalone binaries as artifacts

### LINKED ISSUE #742: Provide Windows and macOS artifacts that don't require installation
This is necessary for bundling in the WPILib installer. The .exes and .apps in the target folder in CI should be sufficient on Windows and macOS respectively. The Linux AppImage is fine to distribute as-is.

Since we have both the app and CLI, we should upload both into one artifact .zip; the WPILib installer builder will extract them.

### PR DESCRIPTION
**Pull request by calcmogul**
Fixes #742.

### CODE REVIEW

**Comment by calcmogul** [comment]
Main questions I have about this still are whether the standalone binaries work on Windows and macOS, and whether I modified the release workflow correctly.

File: .github/workflows/choreo.yml [hunk 1]
```diff
@@ -83,43 +83,33 @@ jobs:
...
           SCCACHE_GHA_ENABLED: true
 
-      - name: Upload bundle (Windows x86)
+      - name: Package artifacts (Windows x86_64)
         if: matrix.artifact-name == 'Windows-x86_64'
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{ matrix.artifact-name }}
-          path: target/x86_64-pc-windows-msvc/release/bundle/nsis/*.exe
+        run: zip -r ${{ matrix.artifact-name }}.zip . -i target/x86_64-pc-windows-msvc/release/choreo.exe target/x86_64-pc-windows-msvc/release/choreo-cli.exe target/x86_64-pc-windows-msvc/release/bundle/nsis/*.exe
```
Comment (Gold856) [inline .github/workflows/choreo.yml:88]:
```suggestion
        run: Compress-Archive -DestinationPath ${{ matrix.artifact-name }}.zip . -Path target/x86_64-pc-windows-msvc/release/choreo.exe,target/x86_64-pc-windows-msvc/release/choreo-cli.exe,target/x86_64-pc-windows-msvc/release/bundle/nsis/*.exe
```

File: .github/workflows/choreo.yml [hunk 2]
```diff
@@ -83,43 +83,33 @@ jobs:
...
 
-      - name: Upload bundle (Windows ARM)
+      - name: Package artifacts (Windows aarch64)
         if: matrix.artifact-name == 'Windows-aarch64'
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{ matrix.artifact-name }}
-          path: target/aarch64-pc-windows-msvc/release/bundle/nsis/*.exe
+        run: zip -r ${{ matrix.artifact-name }}.zip . -i target/aarch64-pc-windows-msvc/release/choreo.exe target/aarch64-pc-windows-msvc/release/choreo-cli.exe target/aarch64-pc-windows-msvc/release/bundle/nsis/*.exe
```
Comment (Gold856) [inline .github/workflows/choreo.yml:92]:
```suggestion
        run: Compress-Archive -DestinationPath ${{ matrix.artifact-name }}.zip . -Path target/aarch64-pc-windows-msvc/release/choreo.exe,target/aarch64-pc-windows-msvc/release/choreo-cli.exe,target/aarch64-pc-windows-msvc/release/bundle/nsis/*.exe
```

**Review by Gold856 — CHANGES_REQUESTED** [review]

File: .github/workflows/choreo.yml [hunk 3]
```diff
@@ -83,43 +83,32 @@ jobs:
...
           SCCACHE_GHA_ENABLED: true
 
-      - name: Upload bundle (Windows x86)
+      - name: Package artifacts (Windows x86_64)
         if: matrix.artifact-name == 'Windows-x86_64'
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{ matrix.artifact-name }}
-          path: target/x86_64-pc-windows-msvc/release/bundle/nsis/*.exe
+        run: Compress-Archive -DestinationPath ${{ matrix.artifact-name }}.zip . -Path target/x86_64-pc-windows-msvc/release/choreo.exe,target/x86_64-pc-windows-msvc/release/choreo-cli.exe,target/x86_64-pc-windows-msvc/release/bundle/nsis/*.exe
```
Comment (Gold856) [inline .github/workflows/choreo.yml:88]:
```suggestion
        run: Compress-Archive -DestinationPath ${{ matrix.artifact-name }}.zip -Path target/x86_64-pc-windows-msvc/release/choreo.exe,target/x86_64-pc-windows-msvc/release/choreo-cli.exe,target/x86_64-pc-windows-msvc/release/bundle/nsis/*.exe
```

File: .github/workflows/choreo.yml [hunk 4]
```diff
@@ -83,43 +83,32 @@ jobs:
...
 
-      - name: Upload bundle (Windows ARM)
+      - name: Package artifacts (Windows aarch64)
         if: matrix.artifact-name == 'Windows-aarch64'
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{ matrix.artifact-name }}
-          path: target/aarch64-pc-windows-msvc/release/bundle/nsis/*.exe
+        run: Compress-Archive -DestinationPath ${{ matrix.artifact-name }}.zip . -Path target/aarch64-pc-windows-msvc/release/choreo.exe,target/aarch64-pc-windows-msvc/release/choreo-cli.exe,target/aarch64-pc-windows-msvc/release/bundle/nsis/*.exe
```
Comment (Gold856) [inline .github/workflows/choreo.yml:92]:
```suggestion
        run: Compress-Archive -DestinationPath ${{ matrix.artifact-name }}.zip -Path target/aarch64-pc-windows-msvc/release/choreo.exe,target/aarch64-pc-windows-msvc/release/choreo-cli.exe,target/aarch64-pc-windows-msvc/release/bundle/nsis/*.exe
```

**Review by Gold856 — CHANGES_REQUESTED** [review]

**Comment by Gold856** [comment]
Can confirm Windows x86_64 works.

**Pull request merged**